### PR TITLE
JESI-2870

### PR DESCRIPTION
**Pull request by axrs**
[JESI-2870](https://jesims.atlassian.net/browse/JESI-2870)

https://github.com/jesims/jesi/pull/932
https://github.com/jesims/jesi-web/pulls
https://github.com/jesims/url/pull/2

### CODE REVIEW

File: backpack.sh [hunk 1]
```diff
@@ -173,6 +174,26 @@ release () {
...
 }
 
+compare-file-from-master() {
+	local branch;
+	if [[ -n $CIRCLE_BRANCH ]];then
+		branch=$CIRCLE_BRANCH
+	else
+		branch=$(git rev-parse --abbrev-ref HEAD)
+	fi
+  echo $(git --no-pager diff --name-only $branch..origin/master -- $1)
```
Comment (axrs) [inline backpack.sh]: bash is tab indented.

File: test/io/jesi/backpack/collection_test.cljc [hunk 2]
```diff
@@ -390,3 +390,162 @@
...
+    (testing "is a function"
+      (is (fn? bp/reduce-leaves))
+
+      (testing "that traverses over a collection, reducing over the leaves"
+
+        (testing "with an init")
+          ;TODO
+
+        (testing "with a leaf predicate")))))
+          ;TODO
```
Comment (axrs) [inline test/io/jesi/backpack/collection_test.cljc]: Make all TODO tests TODONE

File: src/io/jesi/backpack/fn.cljc [hunk 3]
```diff
@@ -78,3 +78,28 @@
...
+
+(def ^{:arglists '([pred & more])
+       :doc      "Higher order `and`.
+       Takes any number of predicates and returns a function that takes a value
+       and returns true if ALL individual predicates return true, else return false."}
+  and-fn
+  (partial apply-predicates every?))
+
+(def
+  ^{:arglists '([pred & more])
```
Comment (BadAlgorithm) [inline src/io/jesi/backpack/fn.cljc]: This probably doesn't need to be on a new line

File: src/io/jesi/backpack/cache.cljc [hunk 4]
```diff
@@ -0,0 +1,91 @@
+(ns io.jesi.backpack.cache
+  (:refer-clojure :exclude [get set])
+  (:require
+    [io.jesi.backpack.macros :refer [reify-ifn]])
+  (:require
```
Comment (AndreTheHunter) [inline src/io/jesi/backpack/cache.cljc]: Combine with above require

File: src/io/jesi/backpack/random.cljc [hunk 5]
```diff
@@ -26,7 +26,8 @@
 
 (def ^:private char-range (range 97 123))                   ;a-z
 
-(defn character []
+(defn character
+  []
```
Comment (AndreTheHunter) [inline src/io/jesi/backpack/random.cljc]: Join lines

**Pull request merged**